### DISCUSSION TITLE
Clear Command for InputCharsetItem

### DISCUSSION THREAD

**Discussion by FranziMermann** [forntoh/LcdMenu#274] [Q&A] [answered]
Great project, thank you very much. I would like to incorporate the ItemInputCharset in my project. 

But for me there seems to be a problem with the clear() method in InputItem. value is set to (char*)"".

' void clear(MenuRenderer* renderer) {

        value = (char*)"";

        draw(renderer);

        renderer->drawBlinker();
 
    }
`
 but the substring helper function in the called function draw(MenuRenderer*) will be called with start greater than 0 if view is greater than 0 and hence garbage is copied to the destination string.

`    void draw(MenuRenderer* renderer) override {

        const uint8_t viewSize = getViewSize(renderer);

        char* vbuf = new char[viewSize + 1];

        substring(value, view, viewSize, vbuf);

        vbuf[viewSize] = '\0';

        renderer->drawItem(text, vbuf);

        delete[] vbuf;

    }`

How do I fix this? Or am I wrong?

Thank You
Frank

**Comment by FranziMermann** [comment, marked as answer]
Sorry, seems that my platformio installation did not pull the latest version. Seems to be fixed in 5.3.1

**Reply by forntoh** [reply]
Glad you found a solution, I've not maintained the lib in 3 months now as I am very busy, there's a lot of things on the main branch which has not be put into a release yet, you can also pull the main branch and test.

**Reply by FranziMermann** [reply]
Hi Thomas, 

Pulled the main branch and sadly the problem persisted. It was to late yesterday, for me to realize. I had to insert a view=0; in the clear(MenuRenderer*) method and now it seemed to work.

Frank

**Reply by forntoh** [reply]
It is a bug, thanks for catching that, fixed in #277
